lib/core/test/unit/Cardano/Wallet/Primitive/Types/StateDeltaSeqSpec.hs

```diff
@@ -0,0 +1,654 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
```

Comment:
Yup! I would be like that. I'm in favor of [PatternSynonyms](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/pattern_synonyms.html), but `ViewPatterns` just mixes up the order of arguments in a confusing way without providing much benefit in terms of abstraction capabilities.

Reply:
@Unisay 

We do use `ViewPatterns` quite a bit throughout our codebase. (Around 15 modules already use it.)

Though I'd definitely be happy to adopt `PatternSynonyms` instead of `ViewPatterns` in future PRs!

Would you be happy with the following proposal:

Proposal:
1. Keep the current usage as it is (for this PR).
2. Create a follow-on PR that replaces all usages of `ViewPatterns` with `PatternSynonyms` (where feasible). That PR can also update other modules not included within the scope of this PR.

The proposal would unblock merging of this PR, and allow the replacement work to be prioritised independently. I'd be delighted to do it as a gardening task. 👍🏻

Reply:
> The proposal would unblock merging of this PR, and allow the replacement work to be prioritised independently. I'd be happy to do it as a gardening task. 👍🏻

As long as `ViewPatterns` goes away, I'm happy.  😂

Reply:
@HeinrichApfelmus wrote:
> As long as ViewPatterns goes away, I'm fine with that. :joy

Okay 😄. I've created a ticket for this: https://input-output.atlassian.net/browse/ADP-2230

My proposal is:
- For this PR, we keep the current usage as it is.
- After this PR is merged, I will create a **follow-on** PR that addresses ADP-2230 for the whole wallet. There are only 15 modules; it shouldn't take too long to replace all the usages with `PatternSynonyms`, or something else (on a case-by-case basis).

@Unisay, are you also okay with this plan?